Source/PINOperationQueue.h

```diff
@@ -21,19 +21,75 @@ PINOP_SUBCLASSING_RESTRICTED
 @interface PINOperationQueue : NSObject
 
 - (instancetype)init NS_UNAVAILABLE;
+
+/**
+ * Initializes and returns a newly allocated operation queue with the specified number of maximum concurrent operations.
+ *
+ * @param maxConcurrentOperations The maximum number of queued operations that can execute at the same time.
+ *
+ */
 - (instancetype)initWithMaxConcurrentOperations:(NSUInteger)maxConcurrentOperations;
+
+/**
+ * Initializes and returns a newly allocated operation queue with the specified number of maximum concurrent operations and the concurrent queue they will be scheduled on.
+ *
+ * @param maxConcurrentOperations The maximum number of queued operations that can execute at the same time.
+ * @param concurrentQueue The operation queue to schedule concurrent operations
+ *
+ */
 - (instancetype)initWithMaxConcurrentOperations:(NSUInteger)maxConcurrentOperations concurrentQueue:(dispatch_queue_t)concurrentQueue NS_DESIGNATED_INITIALIZER;
+
+/**
+ * Returns the shared instance of the PINOperationQueue class.
+ */
 + (instancetype)sharedOperationQueue;
 
+/**
+ * Adds the specified operation object to the receiver.
+ *
+ * @param operation The operation object to be added to the queue.
+ *
+ */
 - (id <PINOperationReference>)addOperation:(dispatch_block_t)operation;
+
+/**
+ * Adds the specified operation object to the receiver.
+ *
+ * @param operation The operation object to be added to the queue.
+ * @param priority The execution priority of the operation in an operation queue.
+ *
+ * @discussion
+ */
 - (id <PINOperationReference>)addOperation:(dispatch_block_t)operation withPriority:(PINOperationQueuePriority)priority;
+
+/**
+ * Adds the specified operation object to the receiver.
+ *
+ * @param operation The operation object to be added to the queue.
+ * @param priority The execution priority of the operation in an operation queue.
+ * @param identifier A string that identifies the operations that can be coalesced
+ * @param coalescingData  
```

Comment:
@nguyenhuy Can you let me know what's best to write for `identifier`, `calescingData` and `dataCoalescingBlock`?